CRM/Dedupe/BAO/DedupeRuleGroup.php

```diff
@@ -14,12 +14,19 @@
  * @package CRM
  * @copyright CiviCRM LLC https://civicrm.org/licensing
  */
+use Civi\Core\HookInterface;
 
 /**
  * The CiviCRM duplicate discovery engine is based on an
  * algorithm designed by David Strauss <david@fourkitchens.com>.
  */
-class CRM_Dedupe_BAO_DedupeRuleGroup extends CRM_Dedupe_DAO_DedupeRuleGroup {
+class CRM_Dedupe_BAO_DedupeRuleGroup extends CRM_Dedupe_DAO_DedupeRuleGroup implements \Symfony\Component\EventDispatcher\EventSubscriberInterface {
+
+  public static function getSubscribedEvents(): array {
```

Comment:
This shouldn't be needed one switched to using the hookInterface

Reply:
how do you specify weight with hook interface?

Reply:
You're right @eileenmcnaughton if you want to specify priority level you need to use `getSubscribedEvents()`